pkg/utils/secrets/token.go

```diff
@@ -0,0 +1,37 @@
+package secrets
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+)
+
+var (
+	chars        = []rune("ABCDEFGHIJKLMNPQRSTUVWXYZabcdefghjkmnpqrstuvwxyz1234567890")
+	specialChars = []rune("#%&()*+-<=>?@[]^_{}")
+)
+
+// CreateToken creates a variable length random token that can be used for one time
+// passwords.
+func CreateToken(length int) string {
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+	var b strings.Builder
+	for i := 0; i < length; i++ {
+		if random.Float64() <= 0.90 {
+			b.WriteRune(chars[random.Intn(len(chars))])
+		} else {
+			b.WriteRune(specialChars[random.Intn(len(specialChars))])
+		}
+	}
+	return b.String()
+}
+
+// ValidateToken checks if a token contains any invalid characters.
+func ValidateToken(token string) bool {
+	for _, c := range token {
+		if !strings.ContainsRune(string(chars), c) && !strings.ContainsRune(string(specialChars), c) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Copied directly from GDS

Reply:
Doh, not sure why we didn't use `crypto/rand` in GDS - but that's probably a problem. Would you mind copying the  `quarterdeck/keygen` package instead?

https://github.com/rotationalio/ensign/blob/main/pkg/quarterdeck/keygen/keygen.go

(You can use it directly if you'd like, but copying it into a secrets utils is fine; I'd like them decoupled so that we can change the way that Quarterdeck issues api keys; e.g. either use the keygen package directly or port it to utils so quarterdeck does not have a dependency on utils). 

I'll create a story to fix GDS.

Reply:
Sure that works for me